src/test/java/com/monumental/services/integrationtest/MonumentServiceIntegrationTests.java

```diff
@@ -0,0 +1,298 @@
+     */
+    @Test
+    public void testMonumentService_Insert_Multiple() {
+        List<Monument> monuments = makeTestMonuments();
+
+        List<Integer> results = this.monumentService.insert(monuments);
+
+        assertEquals(3, results.size());
+
+        for (Integer result : results) {
+            assertNotNull(result);
+        }
+    }
+
+    /**
+     * Test method for integration testing MonumentService.get(record)
+     * First inserts a record into the database then performs a get to verify the records are the same
+     * Uses an H2 in-memory database as to not ruin the actual database
+     */
+    @Test
+    public void testMonumentService_Get_Single() {
+        Monument monument = makeTestMonument();
+
+        Integer id = this.monumentService.insert(monument);
+
+        Monument result = this.monumentService.get(id);
+
+        assertEquals(monument.getTitle(), result.getTitle());
+        assertEquals(monument.getArtist(), result.getArtist());
+    }
+
+    /**
+     * Test method for integration testing MonumentService.getAll(List<record>)
+     * First inserts 3 records into the database then performs a get for only 2 of the records
+     * Checks that only 2 records are returned and that they have the expected attributes
+     * Uses an H2 in-memory database as to not ruin the actual database
+     */
+    @Test
+    public void testMonumentService_GetAll_ListPassed() {
+        ArrayList<Monument> monuments = makeTestMonuments();
+
+        this.monumentService.insert(monuments);
+
+        ArrayList<Integer> idsToGet = new ArrayList<>();
+        idsToGet.add(1);
+        idsToGet.add(3);
+
+        List<Monument> results = this.monumentService.get(idsToGet);
+
+        assertEquals(2, results.size());
+
+        for (Monument result : results) {
+            if (result.getId() == 1) {
+                assertEquals(monuments.get(0).getTitle(), result.getTitle());
+                assertEquals(monuments.get(0).getArtist(), result.getArtist());
+            }
+            else {
+                assertEquals(monuments.get(2).getTitle(), result.getTitle());
+                assertEquals(monuments.get(2).getArtist(), result.getArtist());
+            }
+        }
+    }
+
+    /**
+     * Test method for integration testing MonumentService.getAll(null)
+     * First inserts 3 records into the database then performs a getAll for all 3 records
+     * Checks that 3 records are returned and that they have the expected attributes
+     * Uses an H2 in-memory database as to not ruin the actual database
+     */
+    @Test
+    public void testMonumentService_GetAll_NullPassed() {
+        ArrayList<Monument> monuments = makeTestMonuments();
+
+        this.monumentService.insert(monuments);
+
+        List<Monument> results = this.monumentService.getAll();
+
+        assertEquals(3, results.size());
+
+        for (Monument result : results) {
+            if (result.getId() == 1) {
```

Comment:
I'm being pedantic here, but you could create a map of Id to monument and avoid any if statements here. I do think that you're right that the order in which `getAll` currently makes no sense and should be ordered by _something_, probably Id for now

Reply:
That being said even if it returns ordered by Id I don't think our tests should rely on the order in which results are returned

Reply:
Agreed that at some point we should definitely order `getAll()` by id.

Reply:
Will be fixed in the next commit.